pydrive2/auth.py

```diff
@@ -42,103 +57,6 @@ class RefreshError(AuthError):
     """Access token refresh error."""
 
 
-def LoadAuth(decoratee):
-    """Decorator to check if the auth is valid and loads auth if not."""
-
-    @wraps(decoratee)
-    def _decorated(self, *args, **kwargs):
-        # Initialize auth if needed.
-        if self.auth is None:
```

Comment:
Can is still happen?

Reply:
Is this referring to a decorator to assign an auth object via the decorator?

Reply:
Yep. It's just I see that we are potentially changing the behavior of the library along the way. I would try to preserve it first, list the changes that we do (so that we are aware, users are aware, etc), and change it only if it's actually needed. Or if there are some "ugly" place - change them in a separate PR so that we do things step by step- easier to review then / faster to merge

Reply:
I've added back the LoadAuth decorator but left all the authentication & refresh to the lazy loading properties that I've added. Functionality should still be the same as before

Reply:
I might add some test cases for this auto-population of self.auth as there were no previous tests cases for this. 

However, I strongly feel that this be deprecated in the long term because this decorator does not reuse any existing instance of the GoogleAuth object. We could end up with multiple instances of the same credentials with out-of-sync refreshes